cmd/osv-scanner/callanalysis_parser.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+var stableCallAnalysisStates = map[string]bool{
+	"go":   true,
+	"rust": false,
+}
+
+// Creates a map to record if languages are enabled or disabled for call analysis.
+func createCallAnalysisStates(enabledCallAnalysis []string, disabledCallAnalysis []string) map[string]bool {
+	callAnalysisStates := make(map[string]bool)
+
+	for _, language := range enabledCallAnalysis {
+		callAnalysisStates[language] = true
+	}
+
+	for _, language := range disabledCallAnalysis {
+		callAnalysisStates[language] = false
+	}
+
+	enableAll, containsAll := callAnalysisStates["all"]
+	for language, isStable := range stableCallAnalysisStates {
+		if _, exists := callAnalysisStates[language]; !exists {
```

Comment:
nit: I _think_ this is equivalent, which feels a bit simpler: 

```
if containsAll {
  callAnalysisStates[language] = enableAll
} else if _, exists := callAnalysisStates[language]; !exists {
  callAnalysisStates[language] = isStable
}
```

up to you though as I don't think it's that much simpler